bin/proving-service/src/proxy/worker.rs

```diff
@@ -104,25 +121,52 @@ impl Worker {
             return;
         }
 
+        // If we don't have a status client, try to recreate it
+        if self.status_client.is_none() {
+            let failed_attempts = self.num_failures();
+            self.set_health_status(WorkerHealthStatus::Unhealthy {
+                num_failed_attempts: failed_attempts + 1,
+                first_fail_timestamp: match &self.health_status {
+                    WorkerHealthStatus::Unhealthy { first_fail_timestamp, .. } => {
+                        *first_fail_timestamp
+                    },
+                    _ => Instant::now(),
+                },
+                reason: "No status client available".to_string(),
+            });
```

Comment:
This code (and the code on lines 145 - 156 below) is a bit difficult to follow. Maybe it would make sense to create a small helper method - something like `set_unhealthy_status()`?